### ISSUE TITLE
Require enum values to be explicitly set

### ISSUE THREAD

**Issue by binki** [palantir/tslint#4108]
### Rule Suggestion

**Is your rule for a general problem or is it specific to your development style?**

For a particular project, my enums are saved to a database. When this happens, they become numbers. For my project to continue working after code changes, I require the numeric value of an enum member to stay the same forever—even if I delete, add, or reorder my enum’s members. Since most of the enums in my project are used for this purpose, I would like to use tslint to force me to specify values for all enum members explicitly.

**What does your suggested rule do?**

Requires enums to have values explicitly assigned. For example, disallow the following:

```ts
enum MyEnum {
  A,
  B,
  C,
}
```

Allow this:

```ts
enum MyEnum {
  A = 0,
  B = 1,
  C = 2,
}
```

**List several examples where your rule could be used**

* Serialization of values and later unserialization in an updated version of a program. Having values explicitly set avoids their values changing when the developer reorders the enum values or inserts a new value between existing values or deletes a value.
   * database
   * disk
   * JSON
   * network protocol

**Additional context**

**Comment by JoshuaKGoldberg** [comment]
@binki Although I strongly agree with your use case of serialized values _(having been sternly talked to in code reviews about serialization!)_, this feels like it might be too opinionated for TSLint core. There are valid use cases for not wanting to require enum values to be explicit. For example, internal enums with many members where the values don't matter would be annoying to maintain here.

How about sending this issue to [`tslint-microsoft-contrib`](https://github.com/microsoft/tslint-microsoft-contrib)?

**Comment by JoshuaKGoldberg** [comment]
Closing this for housekeeping but please do comment back here or in tslint-microsoft-contrib if you still want this rule!
